RestaurantsNearMe/Presentation/SplashScreenViewModel.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  SplashScreenViewModel.swift
+//  RestaurantsNearMe
+//
+//  Created by Colin Evans on 2023-11-24.
+//
+
+import Foundation
+import Combine
+import CloudKit
+
+class SplashScreenViewModel: ObservableObject {
+  @Published var isFetching = false
+  @Published var fetchingError: String?
+  
+  private var cancellables = Set<AnyCancellable>()
+  private let cloudKitService: any CloudKitServiceProviding
```

Comment:
This could be provided via a dependency injection framework (as a note, not a requirement or anything)

Reply:
Oh interesting, I used Swinject a bit in the last code base I worked on, but it didn't have anything that advanced. What would using a DI framework achieve here?